src/social-network-provider/twitter.com/ui/custom.ts

```diff
@@ -1,14 +1,40 @@
-import { useState, useEffect, useLayoutEffect } from 'react'
+import { useState, useEffect } from 'react'
+import { ValueRef, MutationObserverWatcher } from '@holoflows/kit'
 import { Theme, createMuiTheme } from '@material-ui/core'
 import { MaskbookDarkTheme, MaskbookLightTheme } from '../../../utils/theme'
 import { SocialNetworkUICustomUI } from '../../../social-network/ui'
+import { useValueRef } from '../../../utils/hooks/useValueRef'
+import { composeAnchorSelector } from '../utils/selector'
 
 type RGB = [number, number, number]
 
+const primaryColorRef = new ValueRef(toRGB([29, 161, 242]))
+const backgroundColorRef = new ValueRef(toRGB([255, 255, 255]))
+
+export function startWatchThemeColor() {
+    function updateThemeColor() {
+        const color = getBackgroundColor(composeAnchorSelector().evaluate()!)
+        const backgroundColor = getBackgroundColor(document.body)
+
+        if (color) {
+            primaryColorRef.value = color
+        }
+        if (backgroundColor) {
+            backgroundColorRef.value = backgroundColor
+        }
+    }
+    new MutationObserverWatcher(composeAnchorSelector())
+        .addListener('onAdd', updateThemeColor)
+        .addListener('onChange', updateThemeColor)
+        .startWatch({
+            childList: true,
+            subtree: true,
```

Comment:
do you need to watch children evaluate the theme change? maybe watch attribute on the body element is enough

Reply:
Em... It's not working when only watching attribute changes.